src/utils/viewer/viewer.cpp

```diff
@@ -732,11 +770,17 @@ void Viewer::mouseScrollEvent(MouseScrollEvent& event) {
   if (!event.offset().y()) {
     return;
   }
-
+  // Use shift for fine-grained zooming
+  float modVal = (event.modifiers() & MouseEvent::Modifier::Shift) ? 1.01 : 1.1;
+  float mod = event.offset().y() > 0 ? modVal : 1.0 / modVal;
+  auto cameraSensor =
+      defaultAgent_->getSensorSuite().getSensors()["rgba_camera"];
+  auto cam = std::static_pointer_cast<esp::sensor::CameraSensor>(cameraSensor);
```

Comment:
```suggestion
  auto& cameraSensor = // what is this type, actually?
      *defaultAgent_->getSensorSuite().getSensors()["rgba_camera"];
  auto& cam = static_cast<esp::sensor::CameraSensor&>(cameraSensor);
```

Less unneeded shared pointer copies ;) Also, could the first `auto` be an explicitly typed-out type? Because I personally can't tell what it is without browsing the docs first.